package.json

```
@@ -953,24 +953,67 @@
+                                            },
+                                            "architecture": {
+                                                "anyOf": [
+                                                    {
+                                                        "type": "string",
+                                                        "enum": [
+                                                            "amd64",
+                                                            "386",
+                                                            "arm64",
+                                                            "arm"
+                                                        ]
+                                                    },
+                                                    {
+                                                        "type": "string"
+                                                    }
+                                                ],
+                                                "description": "%vscode-docker.tasks.docker-build.dockerBuild.platform.architecture%"
+                                            }
+                                        }
                                     },
                                     {
                                         "type": "string",
-                                        "enum": [
-                                            "linux/amd64",
-                                            "linux/386",
-                                            "linux/arm64",
-                                            "linux/arm",
-                                            "darwin/amd64",
-                                            "darwin/386",
-                                            "darwin/arm64",
-                                            "darwin/arm",
-                                            "windows/amd64",
-                                            "windows/386",
-                                            "windows/arm64",
-                                            "windows/arm"
+                                        "anyOf": [
+                                            {
+                                                "type": "string",
+                                                "enum": [
+                                                    "linux/amd64",
+                                                    "linux/386",
+                                                    "linux/arm64",
+                                                    "linux/arm",
+                                                    "darwin/amd64",
+                                                    "darwin/386",
+                                                    "darwin/arm64",
+                                                    "darwin/arm",
```

From [this list](https://go.dev/doc/install/source#environment), it appears only amd64/arm64 are valid architectures for darwin/MacOS, so we shouldn't suggest 386 or arm.

```suggestion
                                                    "darwin/amd64",
                                                    "darwin/arm64",
```

---

Good catch David :grinning: